File: M-Organizing Containers of Balls.py
def organizingContainers(container):
    length = len(container)
    type_balls = [0] * length
    for i in range(length):
        for j in range(length):
            type_balls[j] += container[i][j]
            type_container = [sum(container[i]) for i in range(length)]
    print("type_balls", type_balls)
    print("type_container", type_container)

    type_container.sort()
    type_balls.sort()

    if type_container == type_balls:
        return "Possible"
    else:
        return "Impossible"


# print(
#     organizingContainers(
#         [
#             [1, 3, 1],
#             [2, 1, 2],
#             [3, 3, 3],
#         ]
#     )
#     == "Impossible"
# )

# print(
#     organizingContainers(
#         [
#             [0, 2, 1],
#             [1, 1, 1],
#             [2, 0, 0],
#         ],
#     )
#     == "Possible"
# )

print(
    organizingContainers(
        [[1, 1], [1, 1]],
    )
    == "Possible"
)

File: M-Organizing Containers of Balls.js
/*
 * Complete the 'organizingContainers' function below.
 *
 * The function is expected to return a STRING.
 * The function accepts 2D_INTEGER_ARRAY container as parameter.
 */

function organizingContainers(container) {
  // Write your code here
  const n = container.length;
  const ballTypeCount = Array(n).fill(0);
  const containerSlotCount = Array(n).fill(0);
  for (let cntnr = 0; cntnr < n; cntnr++) {
    for (let ballType = 0; ballType < n; ballType++) {
      const balls = container[cntnr][ballType];
      ballTypeCount[ballType] += balls;
      containerSlotCount[cntnr] += balls;
    }
  }
  const ascOrder = (a, b) => a - b;
  const ballTypeCountStr = JSON.stringify(ballTypeCount.sort(ascOrder));
  const containerSlotCountStr = JSON.stringify(
    containerSlotCount.sort(ascOrder)
  );
  return ballTypeCountStr === containerSlotCountStr ? 'Possible' : 'Impossible';
}
